public/app/features/explore/state/utils.ts

```diff
@@ -120,3 +130,52 @@ export function getResultsFromCache(
   const cacheValue = cacheIdx >= 0 ? cache[cacheIdx].value : undefined;
   return cacheValue;
 }
+
+export function getLogLevelFromLabels(labels: Labels): LogLevel {
+  const labelNames = ['level', 'lvl', 'loglevel'];
+  let levelLabel;
+  for (let labelName of labelNames) {
+    if (labelName in labels) {
+      levelLabel = labelName;
+      break;
+    }
+  }
+  return levelLabel ? getLogLevelFromKey(labels[levelLabel]) : LogLevel.unknown;
+}
+
+/**
+ * Create a new data frame with a single field and values creating by adding field values
+ * from all provided data frames
+ */
+export function aggregateFields(dataFrames: DataFrame[], config: FieldConfig): DataFrame {
```

Comment:
As this is used only in datasource it seems weird it is defined here, is there a reason for this not being in loki namespace?

Reply:
Yeah, it needs a better place. I think this could be shared code as it could be used to aggregate data in Elastic as well.